### PR TITLE
Issue147

### PR DESCRIPTION
**Pull request by umutuz7**

### CODE REVIEW

**Comment by zmelce** [comment]
Code Clarity
Code looks quite clear thanks to variable and method names, indentations and code style is accurate and methods are small and modular.

Bugs
There doesn't seem to be a bug with your code. There is just a minor issue that analyzing your code by using Intellij Code Analyzer, I found typo warnings for just one line in each V1.0.5_create_tags.sql, V1.0.4_create_rating.sql, and V1.0.3_create_violations.sql code pages.  

Missing Features
There are no missing features. Features were supposed to be done in this commit that are done.

Coding Convention
	Variable Names
        Variable names are fine and aligned with Java's naming guidelines.

	Indentation
        Indentation is good and organized according to Java's indentation guidelines. 

	Comments
        Your code is mostly explicit, but it still needs more comments at some places. At least some basic directive comment can be good for especially frontend members of team. You can add more comment explaining what the functions take as input and what it returns.

Tests and documentation
Even though your code does not seem to produce a serious bug, it still needs to be tested. You should complete unit tests for your methods. 
 
Your code, although being quite clear, still needs documentation for others to understand in a valid (Doxygen) format.

Modification
Someone familiar with backend part of this project can easily understand and modify this code but as a member of frontend team, at first view, I encountered some understanding problems due to the lack of comments and documentation.
I don’t see significant suggestion given by IntelliJ Code Analyzer that is running on the methods related with the commit.

Overall, the code is well constructed in general and thanks for your hard work as a member of this team.

**Comment by umutuz7** [comment]
Hi Melce,

Thank you for reviewing my code.
I will consider each item in detail.
For now i share controller endpoints for non-backend developers to understand how requests are made to our system on Amazon.
Here is the link: https://www.getpostman.com/collections/1f8c29b3251aaa24fa46

**Comment by omermtemel** [comment]
Please recreate this pr after merging the otherone and fix the conflicts. especially in migrations.

File: Web-App/src/main/java/org/bounswe2015/group9/universal_access/controllers/TagController.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+    @RequestMapping(value = "/{id}", method = RequestMethod.POST)
+    public ResponseEntity createTag(OAuth2Authentication auth, @RequestBody TagDTO tagDTO,@PathVariable("id") Long id){
+        User user = (User) auth.getUserAuthentication().getPrincipal();
+        try{
+            tagService.createTag(new ViolationDTO(id),tagDTO);
+            return new ResponseEntity(HttpStatus.CREATED);
+        }catch (RuntimeException e){
+            e.printStackTrace();
+            return new ResponseEntity(new ErrorDTO(e.getMessage()), HttpStatus.BAD_REQUEST);
```
Comment (omermtemel) [inline Web-App/src/main/java/org/bounswe2015/group9/universal_access/controllers/TagController.java]: Please add thiss code 
    private ResponseEntity exceptionResponse(RuntimeException e) {
        e.printStackTrace();
        if (e instanceof ConstraintViolationException || e instanceof IllegalArgumentException) {
            return new ResponseEntity<>(new ErrorDTO(e.getMessage()), HttpStatus.BAD_REQUEST);
        } else if (e instanceof RecordNotFoundException) {
            return new ResponseEntity<>(new ErrorDTO(e.getMessage()), HttpStatus.NOT_FOUND);
        } else if (e instanceof ForbiddenProccessException) {
            return new ResponseEntity<>(new ErrorDTO(e.getMessage()), HttpStatus.FORBIDDEN);
        } else {
            return new ResponseEntity<>(new ErrorDTO(e.getMessage()), HttpStatus.INTERNAL_SERVER_ERROR);
        }
    }

and call when you catch an Exception in an Action

**Pull request closed without merging**